sys-freebsd/*10.1*: Changed to use tarball of upstream and bump EAPI=5.

nigoro opened the pull request:
Changed to use tarball of upstream.
@mgorny, You will be able to clean dev space ;-)

Bump to EAPI=5.
QA notice fix.

mgorny commented on eclass/freebsd.eclass:
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
+			local topdir="usr/src/"
+			local extractlist
+			for i in ${EXTRACTONLY} ; do
+				extractlist+=" ${topdir}${i}"
```
I suggest using an array here.

mgorny commented on eclass/freebsd.eclass:
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
+				extractlist+=" ${topdir}${i}"
+			done
+			ebegin "Unpacking parts of ${tarball} to ${WORKDIR}"
+			pushd "${WORKDIR}" > /dev/null
```
||die

mgorny commented on eclass/freebsd.eclass:
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
+			pushd "${WORKDIR}" > /dev/null
+			# https://bugs.gentoo.org/show_bug.cgi?id=550734
+			xzdec -c "${DISTDIR}/${tarball}" | tar -xpf - --strip-components=2 ${extractlist} 2> /dev/null || die "tar extract command failed"
+			popd > /dev/null
```
||die

Also, you can alternatively use:

```
cd "${WORKDIR}" || die
...
cd - || die
```

mgorny commented on eclass/freebsd.eclass:
```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
+			ebegin "Unpacking parts of ${tarball} to ${WORKDIR}"
+			pushd "${WORKDIR}" > /dev/null
+			# https://bugs.gentoo.org/show_bug.cgi?id=550734
+			xzdec -c "${DISTDIR}/${tarball}" | tar -xpf - --strip-components=2 ${extractlist} 2> /dev/null || die "tar extract command failed"
```
`xzdec` doesn't seem like a safe choice. How about `tar -xJpf`? Does that work with BSD tar?

mgorny commented on profiles/thirdpartymirrors:
```diff
@@ -7,7 +7,7 @@ debian		http://ftp.au.debian.org/debian http://ftp.at.debian.org/debian http://f
 fedora		ftp://ftp.uni-klu.ac.at/linux/fedora/linux/core http://fedora.inode.at ftp://ftp.tugraz.at/mirror/redhat-linux/fedora/linux/core ftp://ftp.tugraz.at/mirror/redhat-linux/fedora/linux/extras http://ftp.univie.ac.at/systems/linux/fedora/core http://ftp.univie.ac.at/systems/linux/fedora/extras http://ftp.univie.ac.at/systems/linux/fedora http://ftp.netcraft.com.au/pub/fedora/linux/core http:/ ...
 flightgear	ftp://ftp.de.flightgear.org/pub/fgfs http://mirrors.ibiblio.org/pub/mirrors/flightgear/ftp ftp://mirrors.ibiblio.org/pub/mirrors/flightgear/ftp ftp://ftp.kingmont.com/flightsims/flightgear ftp://ftp.ihg.uni-duisburg.de/Mirrors/ftp.flightgear.org
-freebsd		ftp://ftp.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ar.FreeBSD.org/pub/FreeBSD/ ftp://ftp.au.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.au.FreeBSD.org/pub/FreeBSD/ ftp://ftp.at.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.at.FreeBSD.org/pub/FreeBSD/ ftp://ftp.br.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.br.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ca.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.c ...
+freebsd		http://ftp.FreeBSD.org/pub/FreeBSD/ ftp://ftp.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ar.FreeBSD.org/pub/FreeBSD/ ftp://ftp.au.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.au.FreeBSD.org/pub/FreeBSD/ ftp://ftp.at.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.at.FreeBSD.org/pub/FreeBSD/ ftp://ftp.br.FreeBSD.org/pub/FreeBSD/ ftp://ftp2.br.FreeBSD.org/pub/FreeBSD/ ftp://ftp.ca.F ...
```
I'd appreciate if you didn't mix too many unrelated changes in one PR.

nigoro commented:
I've updated freebsd.eclass.
I dropped xzdec. And, I changed to tar -xJpf.

Before you merge this request, please merge PR https://github.com/gentoo/gentoo-portage-rsync-mirror/pull/139 .

mgorny commented:
Merged, thanks.

mgorny commented:
And cleaned up the dev space :).
